### PR TITLE
Movie row

### PR DESCRIPTION
**Pull request by jturgeon88**

### CODE REVIEW

File: src/components/MovieRow.js [hunk 1]
```diff
@@ -12,21 +12,23 @@ class MovieRow extends Component {
   }
 
   componentDidMount() {
-    this.fetchMovies();
+    const { category } = this.props
+    this.fetchMovies(() => fetchMovies(category.url, category.searchTitle));
```
Comment (b-wils) [inline src/components/MovieRow.js]: You have two functions named fetchMovies(). One from class and one from the api. This makes them hard to distinguish. Consider renaming them or changing your import (import * as movieApi from '../lib/movieApi';).

File: src/components/MovieTile.js [hunk 2]
```diff
@@ -1,11 +1,16 @@
 import React, { Component } from 'react'
 import '../styles/MovieRow.css'
+import styled from 'styled-components'
 
 const MovieTile = ({ movie }) => {
+  const PosterComp = styled.img`
```
Comment (b-wils) [inline src/components/MovieTile.js]: Creating styled components inside a render function will cause a new instance to be allocated on each render. Create new styled components at the top level.

File: src/components/MovieRow.js (hunk 1 above)
Comment (b-wils) [reply src/components/MovieRow.js]: What is the final request path? How does category.searchTitle map to a parameter?

**Review by b-wils — CHANGES_REQUESTED** [review]

File: src/components/MovieTile.js (hunk 2 above)
Comment (jturgeon88) [reply src/components/MovieTile.js]: Done. Moved to top level.

File: src/components/MovieRow.js (hunk 1 above)
Comment (jturgeon88) [reply src/components/MovieRow.js]: Resolved.

File: src/components/Home.js [hunk 3]
```diff
@@ -11,7 +11,7 @@ class Home extends Component {
   render() {
     return (
       <div>
-        {movieCategories.map(category => <MovieRow category={category} />)}
+        {movieCategories.map((category, index) => <MovieRow key={index} category={category} />)}
```
Comment (b-wils) [inline src/components/Home.js]: It is bad practice to use index as a key. The index is not tied to a category, so if the categories change we will get inconsistent results. Ideally we would have an id from the database. Can we use a field from category?

File: src/components/MovieRow.js [hunk 4]
```diff
@@ -28,7 +28,7 @@ class MovieRow extends Component {
     return (
       <div className="movie-row-container">
         <h2 className="movie-row-title">{category.title}</h2>
-        <div className="movie-row">{this.state.movies.map(movie => <MovieTile movie={movie} />)}</div>
+        <div className="movie-row">{this.state.movies.map((movie, index) => <MovieTile key={index} movie={movie} />)}</div>
```
Comment (b-wils) [inline src/components/MovieRow.js]: See above on index as a key

**Review by b-wils — APPROVED** [review]

**Pull request merged**